src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs

```diff
@@ -331,13 +331,16 @@ void UpdateLeftToolbarItems()
 			{
 				UIImage icon = null;
 
-				if (image != null)
+				if (_flyoutBehavior == FlyoutBehavior.Flyout)
 				{
-					icon = result?.Value;
-				}
-				else if (String.IsNullOrWhiteSpace(text) && IsRootPage && _flyoutBehavior == FlyoutBehavior.Flyout)
-				{
-					icon = DrawHamburger();
+					if (image != null)
```

Comment:
Would it make sense to move all the code below this to a separate method?

And then before we call `LoadImage` we can just check the flyout behavior? If it's set to Flyout then we just call the separate method with a `null` icon?

With this path, it seems like we're going to be loading an image that we never use.